File: lib/features/aura/data/models/aura_track.dart
import '../../../../core/utils/locale_helper.dart';

enum AuraCategory { frequency, nature }

class AuraTrack {
  final String id;
  final AuraCategory category;
  final int? hz;
  final String? chakra;
  final String? colorHex;
  final String? icon;
  final String nameEn;
  final String nameEs;
  final String descEn;
  final String descEs;
  final int durationSec;
  final String audioUrl;
  final String license;
  final String? freesoundUser;
  final int sortOrder;

  const AuraTrack({
    required this.id,
    required this.category,
    required this.hz,
    required this.chakra,
    required this.colorHex,
    required this.icon,
    required this.nameEn,
    required this.nameEs,
    required this.descEn,
    required this.descEs,
    required this.durationSec,
    required this.audioUrl,
    required this.license,
    required this.freesoundUser,
    required this.sortOrder,
  });

  String get displayName => LocaleHelper.isSpanish ? nameEs : nameEn;
  String get displayDescription => LocaleHelper.isSpanish ? descEs : descEn;

  factory AuraTrack.fromJson(Map<String, dynamic> json) {
    return AuraTrack(
      id: json['id'] as String,
      category: (json['category'] as String) == 'frequency'
          ? AuraCategory.frequency
          : AuraCategory.nature,
      hz: json['hz'] as int?,
      chakra: json['chakra'] as String?,
      colorHex: json['color_hex'] as String?,
      icon: json['icon'] as String?,
      nameEn: json['name_en'] as String,
      nameEs: json['name_es'] as String,
      descEn: json['desc_en'] as String,
      descEs: json['desc_es'] as String,
      durationSec: json['duration_sec'] as int,
      audioUrl: json['audio_url'] as String,
      license: json['license'] as String? ?? 'CC0',
      freesoundUser: json['freesound_user'] as String?,
      sortOrder: json['sort_order'] as int? ?? 0,
    );
  }
}

File: android/app/src/main/kotlin/com/orbix/pixora/data/models/AuraTrack.kt
package com.orbix.pixora.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * AURA wellness track — maps to `aura_tracks` Postgres table.
 *
 * Two categories so far: "frequency" (Solfeggio Hz tones) and "nature"
 * (rain, ocean, etc.). Audio files live in the `aura-audio` Storage bucket;
 * `audio_url` is pre-built server-side and already includes the full URL.
 */
@Serializable
data class AuraTrack(
    val id: String,
    val category: String,
    val hz: Int? = null,
    val chakra: String? = null,
    @SerialName("color_hex") val colorHex: String? = null,
    val icon: String? = null,
    @SerialName("name_en") val nameEn: String? = null,
    @SerialName("name_es") val nameEs: String? = null,
    @SerialName("desc_en") val descEn: String? = null,
    @SerialName("desc_es") val descEs: String? = null,
    @SerialName("duration_sec") val durationSec: Int = 0,
    @SerialName("file_path") val filePath: String,
    @SerialName("audio_url") val audioUrl: String,
    val license: String = "",
    @SerialName("sort_order") val sortOrder: Int = 0,
    @SerialName("created_at") val createdAt: String? = null,
) {
    /** Effective accent color (falls back to a calm purple if null). */
    val accentHex: String get() = colorHex?.takeIf { it.isNotBlank() } ?: "#6B4F8F"

    /** Spanish-by-default for now (MX target). Wire LocaleHelper later. */
    val displayName: String get() =
        nameEs?.takeIf { it.isNotBlank() } ?: nameEn ?: id
    val displayDesc: String get() =
        descEs?.takeIf { it.isNotBlank() } ?: descEn.orEmpty()

    /** Pretty duration: "15:00" or "1:23". */
    val durationLabel: String get() {
        val m = durationSec / 60
        val s = durationSec % 60
        return "%d:%02d".format(m, s)
    }
}
